src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
```

Comment:
Move ternary inside the operator

also to improve readability move conditional to constants like 
```
const children = people.filter(child => {
    const hasMother = people.some(mother => mother.name === child.mother);
    const isMan = child.sex === 'm';
```

Don't forget to use destruction

You can do it!)